### DISCUSSION TITLE
Custom type with pattern validator.

### DISCUSSION THREAD

**Discussion by GalPeleg1** [pydantic/pydantic#7616] [Question]
Hey, I want to declare on types such as EmailStr or Ipv4, but i don’t want to use functional validation as they won’t run using json schema.
I’m thinking about something like:
MyNewType = Annotated[str, Pattern(".*something.*")]
I think that inside BaseModel.model_fields in a field annotated with MyNewType, the metadata will hold the pattern.

**Comment by baderdean** [comment]
I think I may have the same question, I want to validate a string that's a domain name this way:
```
DomainName = Annotated[str, StringConstraints(
    pattern=r"^([\w-]+\.)*(\w[\w-]{0,66})\.(?P<tld>[a-z]{2,18})$",
    strict=True
    )]
```

But obviously everything validates when I try to do DomainName("anything that isn't a domain"). I don't want to create a dataclass with a unique field since I need to pass it as a path parameter in my endpoint API... (I'm using FastAPI)

There is something about validation of custom string that I don't get

**Reply by GalPeleg1** [reply]
Can you explain your problem? Because you actually helped me to solve mine 😅

**Reply by baderdean** [reply]
I'm using this type with a FastAPI endpoint and actually, it never says it's wrong, for instance the value "notadomain" is passing...

**Reply by baderdean** [reply]
Ok actually it was a FastAPI issue:
https://github.com/tiangolo/fastapi/discussions/10105#discussioncomment-6901418
